misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeJobQueueTest.kt

```diff
@@ -185,6 +204,203 @@ internal class FakeJobQueueTest {
     assertThat(fakeJobQueue.peekJobs(GREEN_QUEUE)).isEmpty()
     assertThat(fakeJobQueue.peekJobs(RED_QUEUE)).isEmpty()
   }
+
+  @Test
+  fun considerDeliveryDelaysInFakeJob() {
```

Comment:
there’s a lot happening in this test. Would you consider separate tests for each of the interesting assertions here?

jobsDoNotStartUntilDeliveryDelayElapses()
multipleJobsMayBeReadyConcurrently()

This test is exercising a lot of code but not without much direction. I’d prefer more small tests that each describe an invariant and test only that.

Reply:
Broke down on three tests:
jobsDoNotStartUntilDeliveryDelayElapses
jobsStartOnDeliveryDelayPassed
jobsStartAfterDeliveryDelayPassed